backend/src/entity/SavedPromotion.ts

```diff
@@ -0,0 +1,41 @@
+import {Entity, ManyToOne, CreateDateColumn, PrimaryColumn, Index} from "typeorm";
+import {User} from "./User";
+import {Promotion} from "./Promotion";
+
+/*
+* Represents a user who saved a promotion
+* * the join/association table which connects User and Promotion
+* * contains custom properties
+* */
+@Entity()
+export class SavedPromotion {
```

Comment:
This is a join/association table that has custom properties. Now whenever we add a user adds a saved promotion, the column `dateSaved` automatically generates the timestamp of when the user saved.

See https://typeorm.io/#/many-to-many-relations/many-to-many-relations-with-custom-properties